src/core/server/opensearch/version_check/ensure_opensearch_version.ts

```diff
@@ -36,16 +36,44 @@
  */
 
 import { timer, of, from, Observable } from 'rxjs';
-import { map, distinctUntilChanged, catchError, exhaustMap } from 'rxjs/operators';
+import { map, distinctUntilChanged, catchError, exhaustMap, mergeMap } from 'rxjs/operators';
+import { get, forEach } from 'lodash';
 import {
   opensearchVersionCompatibleWithOpenSearchDashboards,
   opensearchVersionEqualsOpenSearchDashboards,
 } from './opensearch_opensearch_dashboards_version_compatability';
 import { Logger } from '../../logging';
 import type { OpenSearchClient } from '../client';
 
+export const getNodeId = async (internalClient: OpenSearchClient): Promise<string | null> => {
+  try {
+    const state = await internalClient.cluster.state({
+      filter_path: ['nodes.*.attributes.cluster_id'],
+    });
+    // Aggregate different cluster_ids from the OpenSearch nodes
+    const clusterIdSet = new Set();
+    let defaultClusterId = 0;
+
+    // if attributes.cluster_id is missing, assign the monotonically increasing default cluster_id for each node
+    forEach(state.body.nodes, (node: any) => {
+      clusterIdSet.add(get(node, 'attributes.cluster_id', defaultClusterId++));
+    });
```

Comment:
Since we only care about whether or not all of the nodes have the same cluster_id, why not simplify the logic and return immediately if we find any nodes that don't have the same cluster_id as the first node? I don't see a reason to have a Set or the `defaultClusterId` variable here.

Reply:
We need both of those because we're not just checking if `cluster_id` is different but also handling the edge case where the node attribute is not found/present

Reply:
We're treating them as if they have the same cluster_id by using the logic: If the nodes are missing the `cluster_id` then we're incrementing the defaultClusterId and assigning it to set. We can't actually create the node attribute and assign it to the nodes because that has to be done on OpenSearch side

Reply:
Got it. Then can't we simplify the logic to something like this?

```typescript
const sharedClusterId = state.body.nodes.length > 0 ? get(state.body.nodes[0], 'attributes.cluster_id', null) : null;
return (sharedClusterId === null || state.body.nodes.find(node => sharedClusterId !== get(node, 'attributes.cluster_id', null))) 
  ? null : '_local';
```

Reply:
^ does this logic work or no?

Reply:
This logic doesn't seem to handle the missing node attribute case unless I missed it

Reply:
Oh, I thought it did... If `attributes.cluster_id` is missing on any of the nodes, the default value returned from `get` will be null which will not match the `sharedClusterId` (set from the first node), and then `null` will be returned. Otherwise if all of the nodes have the attribute set and they're all the same, `_local` is returned. I might still be misunderstanding the expectations though.

Reply:
I thought we couldn't mutate the node attributes? Where else are those default values being used outside of this function?

Reply:
Updated logic